crates/locutus-node/src/operations/subscribe.rs

```diff
@@ -268,27 +294,60 @@ where
             };
 
             if !op_storage.ring.has_contract(&key) {
-                //FIXME: should try forward to someone else who may have it first
-                // this node does not have the contract, return a void result to the requester
                 log::info!("Contract {} not found while processing info", key);
-                return Ok(return_err());
-            }
+                log::info!("Trying to found the contract from another node");
 
-            if let Err(_) = op_storage.ring.add_subscriber(key.clone(), subscriber) {
-                // max number of subscribers for this contract reached
-                return Ok(return_err());
-            }
+                let initial_htl = op_storage.ring.max_hops_to_live + 1;
+                let new_requester = target;
+                let new_target = op_storage.ring.closest_caching(&key, initial_htl, &[])[0];
```

Comment:
the  second argument is the number of peers you want returned, in this case it would be 1.